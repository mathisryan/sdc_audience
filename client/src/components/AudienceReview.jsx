import React from 'react';

const AudienceReview = (props) => {
  const row = [];
  for (let i = 0; i < props.reviews.length; i += 2) {
    row.push(props.reviews.slice(i, i + 2));
  }
  const reviews = row.map( (rows, i) => (
    <div className="row" key={i}>
      {rows.map( (audienceReview) => (
      <div className="review-wrapper col p-3" key={audienceReview.id}>  
        <div className="review">
          <div className="review-star">
          {(() => {
            const stars = [];
            for (let i = 0; i < audienceReview.stars; i++) {
              stars.push(<i className="far fa-star" key={i}></i>);
            }
            if (audienceReview.stars % 1 !== 0) {
              stars.push(<i className="far fa-star-half" key={audienceReview.stars + 1}></i>);
            }
            return stars;
          })()}
          </div>
          <div className="review-userReview">{audienceReview.review}</div>
        </div>
        <div className="review-user">
          <i className="far fa-user fa-2x"></i>
          <div>{audienceReview.username}</div>
        </div>
      </div>)
      )}
    </div>
  ));

  return (
  <div className="review-container">
    <div className="audienceReviewHeader row p-3">
      <div className='header-movietitle'>Audience Reviews for <em>{props.title}</em></div>
    </div>
    <div className="reviews">{reviews}</div>
    <div className="viewAll">View All</div>
  </div>
  )
}

export default AudienceReview;