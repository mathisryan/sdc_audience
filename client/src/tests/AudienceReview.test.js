import React from 'react';
import { shallow, mount } from 'enzyme';
import AudienceReview from '../components/AudienceReview';
import './setupTest'

describe('AudienceReview', () => {
  const review = {
    title: "Black Panther",
    reviews: [ { "id": 59, "username": "deal", "review": "Cat ipsum dolor sit amet, touch my tail, i shred your hand purrrr slap owner's face at 5am until human fills food dish. Purr while eating sit in box or steal the warm chair right after you get up damn that dog yet scamper annoy the old grumpy cat, start a fight and then retreat to wash when i lose stand in doorway, unwilling to chose whether to stay in or go", "stars": 0 }, { "id": 60, "username": "do", "review": "dolor sit amet, touch my tail, i shred your hand purrrr slap owner's face at 5am until human fills food dish. Purr while eating sit in box or steal the warm chair right after you get up damn that dog yet scamper annoy the old grumpy cat, start a fight and then retreat to wash when i lose stand in doorway, unwilling to chose whether to stay in or go out. Stare at owner accusingly then wink. Lick face hiss at owner, pee a lot, and meow repeatedly scratch at fence purrrrrr", "stars": 3 }, { "id": 61, "username": "do", "review": "tail, i shred your hand purrrr slap owner's face at 5am until human fills food dish. Purr while eating sit in box or steal the warm chair right after you get up damn that dog yet scamper annoy the old grumpy cat, start a fight and then retreat to wash when i lose stand in doorway, unwilling to chose whether to stay in or go out. Stare at owner accusingly then wink. Lick face hiss", "stars": 3.5 }, { "id": 62, "username": "development", "review": "tail, i shred your hand purrrr slap owner's face at 5am until human fills food dish. Purr while eating sit in box or steal the warm chair right after you get up damn that dog yet scamper annoy the old grumpy cat, start a fight and then retreat to wash when i lose stand in doorway, unwilling to chose whether to stay in or go out. Stare at owner accusingly then wink. Lick face hiss at owner, pee a lot, and meow repeatedly scratch at fence purrrrrr eat muffins and poutine until owner comes back meow all night having their mate disturbing sleeping humans", "stars": 0.5 } ]
  }
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<AudienceReview title={review.title} reviews={review.reviews} debug />);
    expect(component).toMatchSnapshot();
  });
  it('accepts title props', () => {
    const wrapper = mount(<AudienceReview title={review.title} reviews={review.reviews}/>);
    expect(wrapper.props().title).toEqual(review.title);
  });

  it('accepts reviews props', () => {
    const wrapper = mount(<AudienceReview title={review.title} reviews={review.reviews}/>);
    expect(wrapper.props().reviews).toEqual(review.reviews);
  });
  
  it('should have correct title html', () => {
    const wrapper = mount(<AudienceReview title={review.title} reviews={review.reviews}/>);
    expect(wrapper.find('div').at(2).text()).toEqual('Audience Reviews for Black Panther');
  });
});
