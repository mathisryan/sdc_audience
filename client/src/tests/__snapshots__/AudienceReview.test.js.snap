// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudienceReview should render correctly in "debug" mode 1`] = `
<div
  className="review-container"
>
  <div
    className="audienceReviewHeader row p-3"
  >
    <div
      className="header-movietitle"
    >
      Audience Reviews for 
      <em>
        Black Panther
      </em>
    </div>
  </div>
  <div
    className="reviews"
  >
    <div
      className="row"
      key="0"
    >
      <div
        className="review-wrapper col p-3"
        key="59"
      >
        <div
          className="review"
        >
          <div
            className="review-star"
          />
          <div
            className="review-userReview"
          >
            Cat ipsum dolor sit amet, touch my tail, i shred your hand purrrr slap owner's face at 5am until human fills food dish. Purr while eating sit in box or steal the warm chair right after you get up damn that dog yet scamper annoy the old grumpy cat, start a fight and then retreat to wash when i lose stand in doorway, unwilling to chose whether to stay in or go
          </div>
        </div>
        <div
          className="review-user"
        >
          <i
            className="far fa-user fa-2x"
          />
          <div>
            deal
          </div>
        </div>
      </div>
      <div
        className="review-wrapper col p-3"
        key="60"
      >
        <div
          className="review"
        >
          <div
            className="review-star"
          >
            <i
              className="far fa-star"
              key="0"
            />
            <i
              className="far fa-star"
              key="1"
            />
            <i
              className="far fa-star"
              key="2"
            />
          </div>
          <div
            className="review-userReview"
          >
            dolor sit amet, touch my tail, i shred your hand purrrr slap owner's face at 5am until human fills food dish. Purr while eating sit in box or steal the warm chair right after you get up damn that dog yet scamper annoy the old grumpy cat, start a fight and then retreat to wash when i lose stand in doorway, unwilling to chose whether to stay in or go out. Stare at owner accusingly then wink. Lick face hiss at owner, pee a lot, and meow repeatedly scratch at fence purrrrrr
          </div>
        </div>
        <div
          className="review-user"
        >
          <i
            className="far fa-user fa-2x"
          />
          <div>
            do
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
      key="1"
    >
      <div
        className="review-wrapper col p-3"
        key="61"
      >
        <div
          className="review"
        >
          <div
            className="review-star"
          >
            <i
              className="far fa-star"
              key="0"
            />
            <i
              className="far fa-star"
              key="1"
            />
            <i
              className="far fa-star"
              key="2"
            />
            <i
              className="far fa-star"
              key="3"
            />
            <i
              className="far fa-star-half"
              key="4.5"
            />
          </div>
          <div
            className="review-userReview"
          >
            tail, i shred your hand purrrr slap owner's face at 5am until human fills food dish. Purr while eating sit in box or steal the warm chair right after you get up damn that dog yet scamper annoy the old grumpy cat, start a fight and then retreat to wash when i lose stand in doorway, unwilling to chose whether to stay in or go out. Stare at owner accusingly then wink. Lick face hiss
          </div>
        </div>
        <div
          className="review-user"
        >
          <i
            className="far fa-user fa-2x"
          />
          <div>
            do
          </div>
        </div>
      </div>
      <div
        className="review-wrapper col p-3"
        key="62"
      >
        <div
          className="review"
        >
          <div
            className="review-star"
          >
            <i
              className="far fa-star"
              key="0"
            />
            <i
              className="far fa-star-half"
              key="1.5"
            />
          </div>
          <div
            className="review-userReview"
          >
            tail, i shred your hand purrrr slap owner's face at 5am until human fills food dish. Purr while eating sit in box or steal the warm chair right after you get up damn that dog yet scamper annoy the old grumpy cat, start a fight and then retreat to wash when i lose stand in doorway, unwilling to chose whether to stay in or go out. Stare at owner accusingly then wink. Lick face hiss at owner, pee a lot, and meow repeatedly scratch at fence purrrrrr eat muffins and poutine until owner comes back meow all night having their mate disturbing sleeping humans
          </div>
        </div>
        <div
          className="review-user"
        >
          <i
            className="far fa-user fa-2x"
          />
          <div>
            development
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="viewAll"
  >
    View All
  </div>
</div>
`;
